docs/upgrading.md

```diff
@@ -61,6 +61,120 @@ commandHandler->AddResponse(path, kReplyCommandId, replyEncoder);
 
 ```
 
+### Decoupling of `CommandHandlerInterface` from Ember Metadata
+
+CommandHandler Interface was coupled with Ember data in ways that caused bugs if
+not setup correctly, updates were made for decoupling, now this data is provided
+
+With this the interfaces
+[`CommandHandlerInterface::EnumerateGeneratedCommands`](#enumerategeneratedcommands),
+[`CommandHandlerInterface::EnumerateAcceptedCommands`](#enumerateacceptedcommands)
+go through some changes, a SHIM is provided to make the transition simpler
+
+#### Full Changes
+
+##### EnumerateGeneratedCommands
+
+Changed the old callback based iteration, into a ListBuilder based approach for
+the Enumeration of Generated Commands
+
+`CommandHandlerInterface::EnumerateGeneratedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context)`
+becomes
+`CommandHandlerInterface::EnumerateGeneratedCommands(const ConcreteClusterPath & cluster, ListBuilder<CommandId> & builder)`
+
+Changes for implementation
+
+-   old
+
+```cpp
+   for (auto && cmd : { ScanNetworksResponse::Id, NetworkConfigResponse::Id, ConnectNetworkResponse::Id })
+   {
+       VerifyOrExit(callback(cmd, context) == Loop::Continue, /**/);
+   }
+```
+
+-   new
+
+```cpp
+   ReturnErrorOnFailure(
+        builder.AppendElements({ ScanNetworksResponse::Id, NetworkConfigResponse::Id, ConnectNetworkResponse::Id })
+   )
+```
+
+##### EnumerateAcceptedCommands
+
+Expanded `EnumerateAcceptedCommands` Interface to provide the Access Information
+(Attributes and Qualities) using a ListBuilder
+
+`CommandHandlerInterface::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context)`
+becomes
+`CommandHandlerInterface::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, ListBuilder<AcceptedCommandEntry> & builder)`
+
+Changes for implementation:
+
+-   Old
+
+```cpp
+    for (auto && cmd : {
+             Disable::Id,
+             EnableCharging::Id,
+         })
+    {
+        VerifyOrExit(callback(cmd, context) == Loop::Continue, /**/);
+    }
+
+    if (HasFeature(Feature::kV2x))
+    {
+        VerifyOrExit(callback(EnableDischarging::Id, context) == Loop::Continue, /**/);
+    }
+```
+
+-   new
+
+```cpp
+    using QF   = DataModel::CommandQualityFlags;
+    using Priv = Access::Privilege;
+    ReturnErrorOnFailure(builder.AppendElements({
+        { Disable::Id,        QF::kTimed, Priv::kOperate },
+        { EnableCharging::Id, QF::kTimed, Priv::kOperate },
```

Comment:
Should the example be using flags/priv from Metadata.h instead of hardcoding, so people who copy this get the right behavior?